napari/layers/utils/stack_utils.py

```diff
@@ -0,0 +1,188 @@
+
+    layerdata_list = list()
+    for i in range(n_channels):
+        if multiscale:
+            image = [
+                np.take(data[j], i, axis=channel_axis)
+                for j in range(len(data))
+            ]
+        else:
+            image = np.take(data, i, axis=channel_axis)
+        i_kwargs = {}
+        for key, val in kwargs.items():
+            try:
+                i_kwargs[key] = next(val)
+            except StopIteration:
+                raise IndexError(
+                    "Error adding multichannel image with data shape "
+                    f"{data.shape!r}.\nRequested channel_axis "
+                    f"({channel_axis}) had length {n_channels}, but "
+                    f"the '{key}' argument only provided {i} values. "
+                )
+
+        layerdata = (image, i_kwargs, 'image')
+        layerdata_list.append(layerdata)
+
+    return layerdata_list
+
+
+def stack_to_images(stack: Image, axis: int, **kwargs: Dict,) -> List[Image]:
+    """Function to split the active layer into separate layers along an axis
+
+    Parameters
+    ----------
+    stack : napari.layers.Image
+        The image stack to be split into a list of image layers
+    axis : int
+        The axis to split along.
+
+    Returns
+    -------
+    imagelist: list
+        List of Image objects
+    """
+
+    data, meta, _ = stack.as_layer_data_tuple()
+
+    for key in ("contrast_limits", "colormap", "blending"):
+        meta[key] = None
+
+    name = stack.name
+    num_dim = len(data.shape)
+    # n_channels = data.shape[axis]
+
+    if num_dim < 3:
+        warnings.warn(
+            "The image needs more than 2 dimensions for splitting",
+            UserWarning,
+        )
+        return None
+
+    if axis >= num_dim:
+        warnings.warn(
+            "Can't split along axis {}. The image has {} dimensions".format(
+                axis, num_dim
+            ),
+            UserWarning,
+        )
+        return None
+
+    if 'colormap' in kwargs and kwargs['colormap'] is not None:
+        kwargs['colormap'] = itertools.cycle(kwargs['colormap'])
+
+    if meta['rgb']:
+        if axis == (num_dim - 1) or axis == -1:
+            kwargs['rgb'] = False  # split channels as grayscale
+        else:
+            kwargs['rgb'] = True  # split some other axis, remain rgb
+            meta['scale'].pop(axis)
+            meta['translate'].pop(axis)
+    else:
+        kwargs['rgb'] = False
+        meta['scale'].pop(axis)
+        meta['translate'].pop(axis)
+
+    meta.update(kwargs)
+    imagelist = list()
+    layerdata_list = split_channels(data, axis, **meta)
+    for i, tup in enumerate(layerdata_list):
+        idata, imeta, _ = tup
+        layer_name = f'{name} layer {i}'
+        imeta['name'] = layer_name
+
+        image = Image(idata, **imeta)
+
+        imagelist.append(image)
+
+    return imagelist
+
+
+def images_to_stack(
+    images: List[Union[Image, Labels]], axis: int = 0, **kwargs: Dict,
```

Comment:
I guess I see the value in allowing merging `Images` and `Labels` together, but we might need to be more careful inside this function then - for example right now if the first layer is in the list is a labels then our assumptions about propagating metadata will be off.

I think for this PR I'd be fine dropping the `Labels` from the function signature, keeping the assumption of only images and then revisiting this as part of #1442 where we want to add more functionality to convert between images and labels anyway.

Reply:
if you mean dropping the type annotation from the function signature, it A) doesn't change anything about the runtime (as you know) but B) it actually doesn't even change anything about the types! because Labels is an Image.  That is, as far as types are concerned `Union[Image, Labels]` is the same thing as `Image` (but as far as the reader of the signature is concerned ... it's true they might think that this works for labels, which we should at least test if we are going to say that)